src/com/linkedin/parseq/Engine.java

```diff
@@ -204,6 +205,14 @@ public boolean awaitTermination(final int time, final TimeUnit unit) throws Inte
     return _terminated.await(time, unit);
   }
 
+  public boolean addPlanActivityListener(final PlanActivityListener planActivityListener) {
+    return _planActivityListeners.add(planActivityListener);
+  }
+
```

Comment:
The problem is that if we don't make this dynamic, then we impose the order in which component have to be constructed i.e. all listeners have to be constructed before engine. This is OK if we are the only ones adding listeners but this will prevent 3rd party listeners to be added to engine.